### PR TITLE
RPD - You can now select modes individually

### LINKED ISSUE #34356: RPD painting is underpowered
Cannot produce some colors of pipes:
* [x] Gray (aka general)
* [x] Red
* [x] Blue (aka supply)
* [x] Cyan
* [x] Green
* [x] Yellow
* [x] Purple
* [x] Amethyst (aka supplymain)
* [x] Orange
* [x] Violet
* [x] Brown
* [x] Dark

It doesn't work at all the right times:

* [ ] Pipe items (unwrenched) cannot be painted.
* [ ] Non-pipe components can be placed colored but not repainted afterwards, resulting in stuff like unfixable double-purple:

image

### PR DESCRIPTION
**Pull request by nicbn**
[Changelogs]: #
:cl: cacogen and nicbn
tweak: You can now select RPD modes individually.
tweak: RPD autowrenching is now a tgui action rather than an Alt Click option.
/:cl:

Continuation of #36354 

image

Fixes #34356

### CODE REVIEW

**Comment by Supermichael777** [comment]
Dont use Fixes: X if you don't fix the whole bug.

IIRC components should be getting colors from their attached pipes anyhow, not the painter. The painting of the component itself is the bug(because its ugly on sprites that aren't grey-scale).

**Comment by nicbn** [comment]
It actually fixes it as now unwrenched pipes will get deleted.
It doesn't fix being unable to paint unwrenched pipes with the painter however, but the issue says "RPD painting"

**Comment by Supermichael777** [comment]
Do you have all the Colors?

**Comment by Supermichael777** [comment]
Have you tested for unintended messages/behavior?

File: code/game/objects/items/RPD.dm [hunk 1]
```diff
@@ -329,108 +321,111 @@ GLOBAL_LIST_INIT(transit_tube_recipes, list(
...
+	if(istype(A, /obj/item/pipe) || istype(A, /obj/structure/disposalconstruct) || istype(A, /obj/structure/c_transit_tube) || istype(A, /obj/structure/c_transit_tube_pod) || istype(A, /obj/item/pipe_meter))
+		to_chat(user, "<span class='notice'>You start destroying a pipe...</span>")
+		playsound(get_turf(src), 'sound/machines/click.ogg', 50, 1)
+		if(do_after(user, destroy_speed, target = A))
+			activate()
+			qdel(A)
+		return
+
+	//if we're clicking on a pipe that can be painted, paint it
+	else if(istype(A, /obj/machinery/atmospherics/pipe) && !istype(A, /obj/machinery/atmospherics/pipe/layer_manifold))
```
Comment (Supermichael777) [inline code/game/objects/items/RPD.dm]: Hey does this work on heat pipes?

Comment (nicbn) [reply code/game/objects/items/RPD.dm]: yup

**Comment by nicbn** [comment]
Also can you be more specific about which pictures I should post?

**Comment by Supermichael777** [comment]
The ui. also try to reproduce the bugs

**Comment by nicbn** [comment]
The UI is the same but without the mode selector (check the diffs), the bugs are no more

**Comment by MrDoomBringer** [comment]
@nicbn pic of ui?

**Comment by nicbn** [comment]
image

**Comment by nicbn** [comment]
@ShizCalev

**Comment by Barhandar** [comment]
Anything that makes RPD less ridiculously slow and bloated is awesome.

**Comment by duncathan** [comment]
has this addressed any of the issues I mentioned in the original pr?

**Comment by nicbn** [comment]
No, but I didn't get the point of your request. This won't require more precision from the user, the user just won't need to change the RPD mode.

**Comment by duncathan** [comment]
it will take more precision to avoid misclicks when there are multiple pipes on one tile, especially heat exchange pipes

I'd be open to adding a default "automatic" mode that does all three as opposed to removing the modes entirely

**Comment by nicbn** [comment]
Ok I'm going to make it have an auto-paint mode. When on, it will paint placed pipes. When off, clicking placed pipes will place another one on the same tile.

**Comment by nicbn** [comment]
RPD autowrenching is now a tgui action rather than an Alt Click option. You can toggle the paint and eat mode in tgui.

image

**Comment by nicbn** [comment]
UI screenshot at top

**Comment by Barhandar** [comment]
Speaking of which, alt-click action is personally vastly superior to tgui mode because holy fuck tgui is _slow_ for me.

**Pull request merged**